File: app/src/main/java/com/the_roberto/kittengifs/giphy/SearchGifResponse.java
package com.the_roberto.kittengifs.giphy;

import com.google.gson.annotations.SerializedName;

public class SearchGifResponse {
    @SerializedName("data") public Data[] data;
    @SerializedName("pagination") public Pagination pagination;

    public static class Data {
        @SerializedName("images") public Images images;

        public static class Images {
            @SerializedName("original") public Gif original;

            public static class Gif {
                @SerializedName("url") public String url;
                @SerializedName("webp") public String webp;
                @SerializedName("mp4") public String mp4;
            }
        }
    }

    public static class Pagination {
        @SerializedName("total_count") public int totalCount;
    }
}

File: app/src/main/java/com/the_roberto/kittengifs/giphy/SearchGifResponse.kt
package com.the_roberto.kittengifs.giphy

import com.google.gson.annotations.SerializedName

class SearchGifResponse {
    @SerializedName("data") var data: Array<Data>? = null
    @SerializedName("pagination") var pagination: Pagination? = null

    class Data {
        @SerializedName("images") var images: Images? = null

        class Images {
            @SerializedName("original") var original: Gif? = null

            class Gif {
                @SerializedName("url") var url: String? = null
                @SerializedName("webp") var webp: String? = null
                @SerializedName("mp4") var mp4: String? = null
            }
        }
    }

    class Pagination {
        @SerializedName("total_count") var totalCount: Int = 0
    }
}
